fuzz/common/src/vsock.rs

```diff
@@ -0,0 +1,409 @@
+                packet.set_dst_port(*port);
+            }
+            VsockFunction::IsEmpty => {
+                packet.is_empty();
+            }
+            VsockFunction::SetLen { len } => {
+                packet.set_len(*len);
+            }
+            VsockFunction::Type_ => {
+                packet.type_();
+            }
+            VsockFunction::SetType { type_ } => {
+                packet.set_type(*type_);
+            }
+            VsockFunction::Op => {
+                packet.op();
+            }
+            VsockFunction::SetOp { op } => {
+                packet.set_op(*op);
+            }
+            VsockFunction::Flags => {
+                packet.flags();
+            }
+            VsockFunction::SetFlags { flags } => {
+                packet.set_flags(*flags);
+            }
+            VsockFunction::SetFlag { flag } => {
+                packet.set_flag(*flag);
+            }
+            VsockFunction::BufAlloc => {
+                packet.buf_alloc();
+            }
+            VsockFunction::SetBufAlloc { buf_alloc } => {
+                packet.set_buf_alloc(*buf_alloc);
+            }
+            VsockFunction::FwdCnt => {
+                packet.fwd_cnt();
+            }
+            VsockFunction::SetFwdCnt { fwd_cnt } => {
+                packet.set_fwd_cnt(*fwd_cnt);
+            }
+            VsockFunction::SetHeaderFromRaw { bytes } => {
+                let _ = packet.set_header_from_raw(bytes.as_slice());
+            }
+            VsockFunction::_WriteToMem { bytes, addr } => {
+                let _ = mem.write_slice(bytes.as_slice(), GuestAddress(*addr));
+            }
+        }
+    }
+}
+
+// Whether we create a VsockPacket from_rx_virtq_chain or from_tx_virtq_chain
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq)]
+pub enum InitFunction {
+    FromRX,
+    FromTX,
+}
+
+/// Input generated by the fuzzer for fuzzing vsock_rx and vsock_tx
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct VsockInput {
+    pub pkt_max_data: u32,
+    pub init_function: InitFunction,
+    pub descriptors: Vec<FuzzingDescriptor>,
+    pub functions: Vec<VsockFunction>,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::create_corpus_file;
+    use crate::virtio_queue::DEFAULT_QUEUE_SIZE;
+    use std::io::Write;
+    use virtio_bindings::bindings::virtio_ring::{VRING_DESC_F_NEXT, VRING_DESC_F_WRITE};
+    use virtio_queue::mock::MockSplitQueue;
+    use virtio_queue::Descriptor;
+    use virtio_vsock::packet::VsockPacket;
+    use vm_memory::{Bytes, GuestAddress, GuestMemory, GuestMemoryMmap};
+
+    // Random values to be used by the tests for the header fields.
+    const SRC_CID: u64 = 1;
+    const DST_CID: u64 = 2;
+    const SRC_PORT: u32 = 3;
+    const DST_PORT: u32 = 4;
+    const LEN: u32 = 16;
+    const TYPE: u16 = 5;
+    const OP: u16 = 6;
+    const FLAGS: u32 = 7;
+    const FLAG: u32 = 8;
+    const BUF_ALLOC: u32 = 256;
```

Comment:
I'm slightly confused by the magic numbers for LEN and BUF_ALLOC when the comment implies these are all random values which have otherwise added sequentially. Does that mean there are some bounds for some of these values?

Reply:
No particular reason for having these values, other than they look more like sizes. We can update them if it creates confusion.

Reply:
If I remember correctly I just took these from the existing Vsock Packet tests. I can add some more documentation about it being rather arbitrary.